src/modules/Builder/features/Activities/ActivitiesHeader/ActivitiesHeader.tsx

```diff
@@ -1,37 +1,18 @@
 import { useTranslation } from 'react-i18next';
-import { useFieldArray, useFormContext } from 'react-hook-form';
-import { useNavigate, useParams, generatePath } from 'react-router-dom';
 
 import { StyledBuilderBtn } from 'shared/styles';
-import { StyledHeader, BuilderContainerProps } from 'shared/features';
+import { StyledHeader } from 'shared/features';
 import { Svg } from 'shared/components';
-import { page } from 'resources';
-import { getNewActivity } from 'modules/Builder/pages/BuilderApplet/BuilderApplet.utils';
 
-export const ActivitiesHeader: BuilderContainerProps['Header'] = ({ isSticky, children }) => {
-  const { t } = useTranslation('app');
-  const navigate = useNavigate();
-  const { appletId } = useParams();
-
-  const { control } = useFormContext();
-  const { append: appendActivity } = useFieldArray({
-    control,
-    name: 'activities',
-  });
+import { ActivitiesHeaderProps } from './ActivitiesHeader.types';
 
-  const handleAddActivity = () => {
-    const newActivity = getNewActivity();
-
-    appendActivity(newActivity);
-    navigate(
-      generatePath(page.builderAppletActivityAbout, { appletId, activityId: newActivity.key }),
-    );
-  };
+export const ActivitiesHeader = ({ isSticky, children, headerProps }: ActivitiesHeaderProps) => {
```

Comment:
please avoid usage the headerProps as object with single property `onAddActivity`